### PR TITLE
Handle only one v4 and one v6 ip in a pod update

### PR DESCRIPTION
**Pull request by solarkennedy**
New version of k8s (1.19+) are picky about pods
having multiple ip addresses in their pod status.

Specifically, they only allow one of each.

This change ensures we only update pods with one of
each to be compliant, especially in the case
where we have EIPs.

----

Does anyone know what else uses this field?
I'm guessing spinnaker somehow uses this eventually?

### CODE REVIEW

File: vk/backend/backend.go [hunk 1]
```diff
@@ -336,12 +336,27 @@ func (b *Backend) handleUpdate(ctx context.Context, update runner.Update) {
...
+			b.pod.Status.PodIPs = []v1.PodIP{
+				{IP: update.Details.NetworkConfiguration.IPAddress},
+			}
+		}
+
+		// And now, V6, if we have one
+		// TODO: append a non-eip v6 if we can determine that we have one
+		if update.Details.NetworkConfiguration.EniIPv6Address != "" {
+			b.pod.Status.PodIPs = append(b.pod.Status.PodIPs, v1.PodIP{
+				IP: update.Details.NetworkConfiguration.EniIPv6Address,
```
Comment (solarkennedy) [inline vk/backend/backend.go:358]: Oh crap I mixed up ENI and EIP here. this wrong.

Comment (solarkennedy) [reply vk/backend/backend.go:358]: Wait no, I'm right, sorry. We don't have ipv6 EIPs yet.
What we are calling `EniIPv6Address` is the normal ip on a container.

**Comment by sargun** [comment]
I think Spinnaker uses this. IIRC this factors into Sonar as well.

**Review by sargun — APPROVED** [review]

**Review by rgulewich — APPROVED** [review]

**Comment by corindwyer** [comment]
Yes, the current integration uses annotations to get ip information.
https://github.com/Netflix/titus-control-plane/blob/caee1e51be0bacf14ea7d63ef15814ed1c024c65/titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeUtil.java#L117

**Pull request merged**